backends/events/default/default-events.cpp

```diff
@@ -180,7 +180,7 @@ bool DefaultEventManager::pollEvent(Common::Event &event) {
 		break;
 
 	case Common::EVENT_RETURN_TO_LAUNCHER:
-		if (ConfMan.getBool("confirm_exit")) {
+		if (g_engine && !g_engine->hasFeature(Engine::kSupportsCustomGUI) && ConfMan.getBool("confirm_exit")) {
 			if (_confirmExitDialogActive) {
```

Comment:
That would allow the engine to replace the dialog directly without requiring a feature flag. 🤔

Reply:
The change that you propose looks good enough to me, and it may indeed be safer to stay with a simpler fix so close to the next release. This can always be changed later.